vega_sim/scenario/fuzzed_markets/agents.py

```diff
@@ -123,17 +140,33 @@ def create_fuzzed_amendment(self):
             pegged_offset=self.random_state.normal(loc=0, scale=10),
         )
 
-    def create_fuzzed_submission(self):
-        return self.vega.create_order_submission(
-            market_id=self.market_id,
-            side=self.random_state.choice(
-                a=["SIDE_UNSPECIFIED", "SIDE_BUY", "SIDE_SELL"],
-            ),
-            size=self.random_state.poisson(lam=10),
-            order_type=self.random_state.choice(
-                a=["TYPE_UNSPECIFIED", "TYPE_MARKET", "TYPE_LIMIT"]
-            ),
-            time_in_force=self.random_state.choice(
+    def create_fuzzed_submission(self, vega_state):
+        self.__class__.MEMORY["TRADING_MODE"].append(
```

Comment:
You actually shouldn't need this class funkiness here (and it's a 'feature' of which one needs to be aware, but is useful in this case!) the `MEMORY` dict is shared in any case as it is defined at top level:

![image](https://user-images.githubusercontent.com/702798/226616394-6d4a51ef-5c35-4624-80c2-83ae3c1984f4.png)

Reply:
So this is exactly how I thought it should work originally but it doesn't seem to... not sure if I am missing something painfully obvious but removing the `__class__` results in a unique `MEMORY` for each market and a unique `OUTPUTTED` flag for each agent! So when the environment is finalised I get 20 figures created, 10 of Market A and 10 of Market B! Any ideas?